MoreLinq/GroupPartition.cs

```diff
@@ -0,0 +1,144 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#endregion
+
+namespace MoreLinq
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Diagnostics;
+    using System.Linq;
+
+    static partial class MoreEnumerable
+    {
+        /// <summary>
+        /// Partitions a grouping by Boolean keys into a projection of true
+        /// elements and false elements, respectively.
+        /// </summary>
+
+        public static TResult Partition<T, TResult>(this IEnumerable<IGrouping<bool, T>> source,
+            Func<IEnumerable<T>, IEnumerable<T>, TResult> resultSelector) =>
+            source.Partition(true, false, (t, f, _) => resultSelector(t, f));
+
+        /// <summary>
+        /// Partitions a grouping by nullable Boolean keys into a projection of
+        /// true elements, false elements and null elements, respectively.
+        /// </summary>
+
+        public static TResult Partition<T, TResult>(this IEnumerable<IGrouping<bool?, T>> source,
+            Func<IEnumerable<T>, IEnumerable<T>, IEnumerable<T>, TResult> resultSelector) =>
+            source.Partition(true, false, null, (t, f, n, _) => resultSelector(t, f, n));
+
+        /// <summary>
+        /// Partitions a grouping into a projection of elements matching a key
+        /// and those groups that do not.
+        /// </summary>
+
+        public static TResult Partition<TKey, TElement, TResult>(this IEnumerable<IGrouping<TKey, TElement>> source,
+            TKey key,
+            Func<IEnumerable<TElement>, IEnumerable<IGrouping<TKey, TElement>>, TResult> resultSelector) =>
+            Partition(source, key, null, resultSelector);
+
+        /// <summary>
+        /// Partitions a grouping into a projection of elements matching a key
+        /// and those groups that do not. An additional parameter specifies how
+        /// to compare keys for equality.
+        /// </summary>
+
+        public static TResult Partition<TKey, TElement, TResult>(this IEnumerable<IGrouping<TKey, TElement>> source,
+            TKey key, IEqualityComparer<TKey> comparer,
+            Func<IEnumerable<TElement>, IEnumerable<IGrouping<TKey, TElement>>, TResult> resultSelector) =>
+            PartitionImpl(source, 1, key, default(TKey), default(TKey), comparer, (a, b, c, rest) => resultSelector(a, rest));
+
+        /// <summary>
+        /// Partitions a grouping into a projection of elements matching two
+        /// sets of keys and those groups that do not.
```

Comment:
s/sets of // ? Or instead "elements matching a set of two keys". Same for the 3 key overloads